index.js

```diff
@@ -0,0 +1,33 @@
+const validateLineup = (lineup) => {
```

Comment:
You've got a lot of extra looping in here. A filter inside a for loop is a lot. If you used reduce to reformat the lineup into objects first you'd have a lot less looping to do.